File: Stack/0682.Baseball Game/Solution.java
class Solution {
    public int calPoints(String[] ops) {
        int ans = 0;
        Deque<Integer> stk = new ArrayDeque<>();
        for (String o : ops) {
            int num = 0;
            if ("+".equals(o)) {
                int last = stk.pollLast();
                num += last;
                num += stk.peekLast();
                stk.addLast(last);
            } else if ("D".equals(o)) {
                num = stk.peekLast() * 2;
            } else if ("C".equals(o)) {
                num = -stk.pollLast();
            } else {
                num = Integer.parseInt(o);
            }
            ans += num;
            if (!"C".equals(o)) {
                stk.addLast(num);
            }
        }
        return ans;
    }
}

File: Stack/0682.Baseball Game/Solution.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        ans = 0
        stk = []
        for o in ops:
            num = 0
            if "+" == o:
                num += stk[-2]
                num += stk[-1]
            elif "D" == o:
                num = stk[-1] * 2
            elif "C" == o:
                num = -stk.pop()
            else:
                num = int(o)
            ans += num
            if "C" != o:
                stk.append(num)
        return ans